### PR TITLE
Do not reject FalseClass from tags/values

### PR DESCRIPTION
**Pull request by hennevogel**
false is blank

Loading production environment (Rails 6.0.3.6)
[1] pry(main)> false.blank?
=> true

### CODE REVIEW

**Comment by hennevogel** [comment]
```ruby
[1] pry(main)> blah = false
=> false
[2] pry(main)> blah && blah.blank?
=> false
[3] pry(main)> blah = ''
=> ""
[4] pry(main)> blah && blah.blank?
=> true
[5] pry(main)> blah = []
=> []
[6] pry(main)> blah && blah.blank?
=> true
[7] pry(main)> blah = :hans
=> :hans
[8] pry(main)> blah && blah.blank?
=> false
[9] pry(main)> blah = 'hans'
=> "hans"
[10] pry(main)> blah && blah.blank?
=> false
```

**Review by danidoni — APPROVED** [review]

File: lib/influxdb/rails/tags.rb [hunk 1]
```diff
@@ -9,7 +9,11 @@ def initialize(config:, tags: {}, additional_tags: InfluxDB::Rails.current.tags)
 
       def to_h
         expanded_tags.reject do |_, value|
-          value.blank?
+          # The first condition is not testing for existance, but for FalseClass.
+          # Because FalseClass is blank but should not be removed.
+          # rubocop:disable Style/SafeNavigation
+          value && value.blank?
```
Comment (ChrisBr) [inline lib/influxdb/rails/tags.rb]:
```ruby
irb(main):028:0> {key: nil}.reject { |_, v| v && v.blank? }
=> {:key=>nil}
```

**Review by ChrisBr — CHANGES_REQUESTED** [review]

File: lib/influxdb/rails/tags.rb (hunk 1 above)
Comment (ChrisBr) [reply lib/influxdb/rails/tags.rb]:
```ruby
expanded_tags.reject do |_, value|
  return true unless value

  value.blank?
end
```

File: spec/support/rails5/app.rb [hunk 2]
```diff
@@ -56,8 +56,8 @@ class MetricsController < ApplicationController
   prepend_view_path File.join(__dir__, "..", "views")
 
   before_action do
-    InfluxDB::Rails.current.values = { additional_value: :value }
-    InfluxDB::Rails.current.tags = { additional_tag: :value }
+    InfluxDB::Rails.current.values = { additional_value: :value, empty_array: [], empty_string: "  ", custom_value: %w[a b] }
```
Comment (ChrisBr) [inline spec/support/rails5/app.rb]: We should test for a `nil` value.

Although usually I'm for high level tests, I think in this case I would probably prefer a unit test.

```ruby
it 'removes nil values' do
  expect(Tags.new({ key: nil }).to_h).to be empty
end

it 'removes empty string values' do
  expect(Tags.new({ key: '   ' , string: ' }).to_h).to be empty
end
  
```

File: lib/influxdb/rails/tags.rb (hunk 1 above)
Comment (hennevogel) [reply lib/influxdb/rails/tags.rb:15]:
```ruby
expanded_tags.reject do |_, value|
  return true if value.is_a?(FalseClass)

  value.blank?
end
```

**Comment by hennevogel** [comment]
@ChrisBr made this a bit more explicit and spec'ed out `Value`. Please re-review

File: spec/unit/tags.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
+require "spec_helper"
+
+RSpec.describe InfluxDB::Rails::Tags do
+  let(:config) { InfluxDB::Rails::Configuration.new }
+
+  describe ".to_h" do
+    it "returns TrueClass" do
+      subject = InfluxDB::Rails::Tags.new(config: config, tags: { hans: true })
+      expect(subject.to_h).to a_hash_including(hans: true)
```
Comment (ChrisBr) [inline spec/unit/tags.rb:9]: [nit] IIRC `a_hash_including` / `hash_including` are just alias for `include` which reads a lot nicer in this case.

```suggestion
      expect(subject.to_h).to include(hans: true)
```

https://gist.github.com/JunichiIto/f603d3fbfcf99b914f86

**Review by ChrisBr — APPROVED** [review]

File: spec/unit/tags.rb (hunk 3 above)
Comment (hennevogel) [reply spec/unit/tags.rb:9]: I think they invented `a_hash_including` because it reads nicer and produces better error messages hehe

Comment (hennevogel) [reply spec/unit/tags.rb:9]: https://rspec.info/blog/2014/01/new-in-rspec-3-composable-matchers/#matcher-aliases

**Pull request merged**